OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -117,8 +117,14 @@ public IEnumerable<FrozenActor> FrozenActorsAt(Player viewer, int2 worldPx)
 
 			var i = (worldPx.X / info.BinSize).Clamp(0, cols - 1);
 			var j = (worldPx.Y / info.BinSize).Clamp(0, rows - 1);
+			int xDiff, yDiff;
 			return frozen[viewer][j * cols + i]
-				.Where(kv => kv.Key.IsValid && kv.Value.Contains(worldPx))
+				.Where(kv => kv.Key.IsValid && kv.Value.Contains(worldPx)).OrderBy(kv =>
+					{
+						xDiff = kv.Value.X + kv.Value.Width / 2 - worldPx.X;
+						yDiff = kv.Value.Y + kv.Value.Height / 2 - worldPx.Y;
+						return xDiff * xDiff + yDiff * yDiff;
```

Comment:
You can use ```int2``` to do vector arithmetic in lots of places in this PR. It will save you writing the same line twice all the time.

For example, you could rewrite this as.
```
var diff = new int2(kv.Value.Location) + new int2(kv.Value.Size) / 2 - worldPx;
return diff.LengthSquared;
```